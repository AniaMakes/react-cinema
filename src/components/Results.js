import React from 'react';
import ResultsItem from './ResultsItem';

class Results extends React.Component{
	constructor(props){
		super(props);


	}
	render(){

		const results = this.props.results;

		let resultsComponentsArray = results.map(function(filmInfo){
			return <ResultsItem 
				title={filmInfo.Title}
				year={filmInfo.Year}
				imdbID={filmInfo.imdbID}
				posterURL={filmInfo.Poster}
				key={filmInfo.imdbID}

			/>
		})

		return(
			<section className='results'>
				<h2>Results</h2>

				{resultsComponentsArray}

			</section>
		);
	}
}

export default Results;