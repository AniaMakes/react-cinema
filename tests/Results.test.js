import React from "react";
import Results from "../src/components/Results"
import renderer from 'react-test-renderer'

describe("Results", () => {
  it ("Results is rendered", () => {
    const mock = renderer.create(<Results
      results={[
        {
          "Title": "Flubber",
          "Year": "1997",
          "imdbID": "tt0119137",
          "Type": "movie",
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BZDlkOTA0OTUtOThmMi00NTMyLTg0NDgtZWZiYmRjZWI4MDY4XkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
        },
        {
          "Title": "Son of Flubber",
          "Year": "1963",
          "imdbID": "tt0057518",
          "Type": "movie",
          "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMTM2Zjk4MDgtYjMzYi00NzA3LWIxODktODcwYjVmOWQ1YzU2XkEyXkFqcGdeQXVyNjE5MjUyOTM@._V1_SX300.jpg"
        },
        {
          "Title": "Absent Minded Inventions and the Search for Flubber with Bill Nye the Science Guy",
          "Year": "1997",
          "imdbID": "tt0432245",
          "Type": "movie",
          "Poster": "N/A"
        }
      ]}
    />).toJSON();

    expect(mock).toMatchSnapshot();

  })
})
