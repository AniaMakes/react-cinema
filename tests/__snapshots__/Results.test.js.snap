// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Results is rendered 1`] = `
<section
  className="results"
>
  <h2>
    Results
  </h2>
  <article
    className="results-item"
    id="tt0119137"
  >
    <div
      className="results-wrapper"
    >
      <img
        className="item-poster"
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BZDlkOTA0OTUtOThmMi00NTMyLTg0NDgtZWZiYmRjZWI4MDY4XkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
      />
      <div
        className="item-info-wrapper"
      >
        <h3
          className="item-title"
        >
          Flubber
        </h3>
        <p
          className="item-year"
        >
          1997
        </p>
      </div>
      <button
        onClick={[Function]}
      >
         More 
      </button>
    </div>
    <div
      className="hide-details"
    >
      <p
        className="film-runtime"
      >
         
        <b>
          Runtime:
        </b>
         
         
      </p>
      <p
        className="film-genre"
      >
         
        <b>
           Genre:
        </b>
         
         
      </p>
      <p
        className="film-plot"
      >
         
      </p>
    </div>
  </article>
  <article
    className="results-item"
    id="tt0057518"
  >
    <div
      className="results-wrapper"
    >
      <img
        className="item-poster"
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTM2Zjk4MDgtYjMzYi00NzA3LWIxODktODcwYjVmOWQ1YzU2XkEyXkFqcGdeQXVyNjE5MjUyOTM@._V1_SX300.jpg"
      />
      <div
        className="item-info-wrapper"
      >
        <h3
          className="item-title"
        >
          Son of Flubber
        </h3>
        <p
          className="item-year"
        >
          1963
        </p>
      </div>
      <button
        onClick={[Function]}
      >
         More 
      </button>
    </div>
    <div
      className="hide-details"
    >
      <p
        className="film-runtime"
      >
         
        <b>
          Runtime:
        </b>
         
         
      </p>
      <p
        className="film-genre"
      >
         
        <b>
           Genre:
        </b>
         
         
      </p>
      <p
        className="film-plot"
      >
         
      </p>
    </div>
  </article>
  <article
    className="results-item"
    id="tt0432245"
  >
    <div
      className="results-wrapper"
    >
      <img
        className="item-poster"
        src="N/A"
      />
      <div
        className="item-info-wrapper"
      >
        <h3
          className="item-title"
        >
          Absent Minded Inventions and the Search for Flubber with Bill Nye the Science Guy
        </h3>
        <p
          className="item-year"
        >
          1997
        </p>
      </div>
      <button
        onClick={[Function]}
      >
         More 
      </button>
    </div>
    <div
      className="hide-details"
    >
      <p
        className="film-runtime"
      >
         
        <b>
          Runtime:
        </b>
         
         
      </p>
      <p
        className="film-genre"
      >
         
        <b>
           Genre:
        </b>
         
         
      </p>
      <p
        className="film-plot"
      >
         
      </p>
    </div>
  </article>
</section>
`;
